virttest/qemu_devices/qdevices.py

```diff
@@ -2180,6 +2180,146 @@ def __eq__(self, other):
         return False
 
 
+class QNetDevice(QCustomDevice):
+    def __init__(self, nettype, params=None, legacy=False):
+        """
+        netdev device
+
+        :param nettype: The type of the netdev device.
+        :type nettype: str
+        :param params: The parameters of the netdev device.
+        :type params: dict
+        :param legacy: Whether the netdev device is in legacy mode (-net).
+        :type legacy: bool
+        """
+        super().__init__('netdev', params)
+        if legacy:
+            self.type = 'net'
+        self.set_param('type', nettype)
```

Comment:
From the code implementation, looks like the priority `nettype` is higher than `legacy` mode.
But from my understanding, both of them are mutually exclusive relationships.
Please correct me if I misunderstanding.

Reply:
`nettype` means `tap`, `user` or others, both `-netdev` and `-net` support them.

but after discussed with @luckyh, will use `QNetdev` to handle `-netdev` only, for legacy mode, use QCustomDevice to avoid regressions.

Reply:
> `nettype` means `tap`, `user` or others, both `-netdev` and `-net` support them.
> 
> but after discussed with @luckyh, will use `QNetdev` to handle `-netdev` only, for legacy mode, use QCustomDevice to avoid regressions.

Okay, got it.